src/pages/download/index.astro

```diff
@@ -41,8 +42,17 @@ import "./index.css";
     }
   };
 
+  const handleDownloadButton = () => {
+    const downloadButton = document.querySelector(".download-btn");
+    downloadButton?.addEventListener("click", () => {
+      // @ts-expect-error // Inject deviceId into window
+      generateZIP(window.targetDeviceId);
```

Comment:
Maybe we should re-think how we pass around data in MUP now; using window as a global context seems very fragile.

But that's another issue

Reply:
opened https://linear.app/oursky/issue/MUP-175/restructure-how-data-is-passed-around